x-pack/plugins/observability_solution/observability/public/components/alerts_flyout/alert_flyout_overview/helpers/getGroups.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ALERT_GROUP_FIELD, ALERT_GROUP_VALUE } from '@kbn/rule-data-utils';
+import {
+  apmSources,
+  infraSources,
+} from '../../../../../common/custom_threshold_rule/helpers/get_alert_source_links';
+import { TopAlert } from '../../../..';
+
+interface AlertFields {
+  [key: string]: any;
+}
+
+export const getGroups = (alert: TopAlert) => {
+  const groupsFromGroupFields = alert.fields[ALERT_GROUP_FIELD]?.map((field, index) => {
+    const values = alert.fields[ALERT_GROUP_VALUE];
+    if (values?.length && values[index]) {
+      return { field, value: values[index] };
+    }
+  });
+
+  if (groupsFromGroupFields?.length) return groupsFromGroupFields;
+
+  // Not all rules has group.fields, in that case we search in the alert fields.
+  const matchedSources: Array<{ field: string; value: any }> = [];
```

Comment:
I wonder if we could use this way directly without checking first the groups, as it will be compatible with all rule type